src/rpc/rawtransaction.cpp

```diff
@@ -922,12 +922,84 @@ static RPCHelpMan testmempoolaccept()
         UniValueType(), // VNUM or VSTR, checked inside AmountFromValue()
     });
 
-    if (request.params[0].get_array().size() != 1) {
-        throw JSONRPCError(RPC_INVALID_PARAMETER, "Array must contain exactly one raw transaction for now");
+    UniValue transactions_in = request.params[0].get_array();
+    if (transactions_in.size() > 1) {
+        unsigned int num_txns = transactions_in.size();
+        UniValue result(UniValue::VARR);
+        const CFeeRate max_raw_tx_fee_rate = request.params[1].isNull() ?
+                                             DEFAULT_MAX_RAW_TX_FEE_RATE :
+                                             CFeeRate(AmountFromValue(request.params[1]));
+
+        CTxMemPool& mempool = EnsureMemPool(request.context);
+
+        std::vector<const CTransactionRef> txns;
+        //decode transactions
+        for (unsigned int i = 0; i < num_txns; i++) {
```

Comment:
If you don't need the index `i`, it's better to use a range based loop:

```suggestion
        for (auto& tx_in : transactions_in) {
```

Reply:
eenteresting, I think UniValue might support indexing but not range based, `invalid range expression of type 'UniValue'; no viable 'begin' function available`

Reply:
oh, fascinating!